### PR TITLE
Allow flexible overpass settings

### PR DESCRIPTION
**Pull request by pmartincalvo**
Until now, overpass QL [settings](https://wiki.openstreetmap.org/wiki/Overpass_API/Overpass_QL#Settings) (the first statement in an overpass query) were mostly hardcoded, with only maxsize and timeout being open to change from the high level functions of OSMnx.

In issue https://github.com/gboeing/osmnx/issues/384, the ability to modify the date for the overpass query was requested as a feature. The date in an overpass query is modified in its settings. When I looked into it, I noticed it would be probably wiser to make settings customizable in a similar fashion to the current `custom_filters`, instead of making an hyperspecialized design just for the date.

The proposed change allows the user to feed a customized settings string. If no string is given, the same default values that are now hardcoded will be used, thus not breaking backwards compatibility.

@gboeing , no tests for this yet. I've tried to understand the testing philosophy of OSMnx from the current tests but I haven't seen any testing related to querying Overpass. Could you kindly let me know your thoughts on this?

### CODE REVIEW

File: osmnx/core.py [hunk 1]
```diff
@@ -221,6 +224,11 @@ def osm_net_download(polygon=None, north=None, south=None, east=None, west=None,
     else:
         maxsize = '[maxsize:{}]'.format(memory)
 
+    # use custom settings if delivered, otherwise just the default ones.
+    overpass_settings = '[out:json][timeout:{timeout}]{maxsize}'.format(timeout=timeout, maxsize=maxsize)
```
Comment (gboeing) [inline osmnx/core.py]: this should probably be converted to an if/else pattern rather than an if/overwrite pattern

Comment (pmartincalvo) [reply osmnx/core.py]: Done.

**Comment by pmartincalvo** [comment]
Makes sense. I'll try to chase down all occurrences and apply the same philosophy.

On a side note, right now the default value for the settings is getting hardcoded in the function. If you approve, I would like to move it to the settings file and import it where necessary so there's a single point for modifying if necessary. 

Will wait for your confirmation on that.

**Comment by gboeing** [comment]
> On a side note, right now the default value for the settings is getting hardcoded in the function. If you approve, I would like to move it to the settings file and import it where necessary so there's a single point for modifying if necessary.

I think that would make sense for usability and consistency with the rest of the package.

**Comment by pmartincalvo** [comment]
Footprints and POIs now also allow for custom settings. I have also set the default setting string in settings.py as discussed.

Unless you think the same needs to be applied somewhere else, I would say only testing is pending.

**Comment by pmartincalvo** [comment]
Tests done. Do let me know if any other changes are required.

**Comment by gboeing** [comment]
@pmartincalvo thanks!

**Comment by gboeing** [comment]
closes #384

**Comment by gboeing** [comment]
Note that this `custom_settings` parameter has been deprecated. It is instead now easily accessible via the ox.config() function and settings module as `overpass_settings`. See #492.

**Pull request merged**